File: CloudMusic/src/main/java/com/ken/android/CloudMusic/Service/MyDialog.java
package com.ken.android.CloudMusic.Service;

import android.app.Dialog;
import android.content.Context;
import android.view.View;
import android.view.Window;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListView;

import com.ken.android.CloudMusic.R;

import java.util.ArrayList;


public class MyDialog extends Dialog implements AdapterView.OnItemClickListener {

    private ListView mListView;
    private ArrayAdapter mAdapter;
    private ArrayList<String> mList;

    private Button okBtn;
    private Button cancleBtn;
    private int buttonNum=2;

    //接口对象
    private OnBtnClickListemer mOnOKBtnClickListemer;

    public MyDialog(Context context, int buttonNum) {
        super(context, R.style.dialog);
        this.buttonNum=buttonNum;
        Window window = this.getWindow();

        this.requestWindowFeature(Window.FEATURE_NO_TITLE);
        window.setContentView(R.layout.my_dailog_layout);
        initView(window);

    }

    //构造函数
    public MyDialog(Context context) {
        super(context, R.style.dialog);

        Window window = this.getWindow();

        this.requestWindowFeature(Window.FEATURE_NO_TITLE);
        window.setContentView(R.layout.my_dailog_layout);
        initView(window);

    }

    private void initView(Window window){
      //  okBtn = (Button) window.findViewById(R.id.ok_btn);
       // cancleBtn = (Button) window.findViewById(R.id.cancle_btn);
        if(buttonNum==1){
            cancleBtn.setVisibility(View.GONE);
        }
        mListView= (ListView) window.findViewById(R.id.id_dialog_list);
        mListView.setOnItemClickListener(this);
        mList=new ArrayList<String>();
        mList.add(0,"我的收藏");
        mList.add(1,"东方幻想");
        mList.add(2,"动漫收藏");
        mAdapter=new ArrayAdapter(window.getContext(),android.R.layout.simple_list_item_1,mList);
        mListView.setAdapter(mAdapter);
    }

    public void showDiglog() {
        show();
    }


    public void setBtnText(String leftText,String rightText){
        okBtn.setText(leftText);
        cancleBtn.setText(rightText);
    }

    public void setBtnText(String leftText){
        okBtn.setText(leftText);
    }
    public void setBtnClick(final OnBtnClickListemer onBtnClickListemer){


        okBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBtnClickListemer.OnOKBtnClick();//接口对象的方法，需要在实例化MyDialog是实现该回调方法
            }
        });
        cancleBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dismiss();
                onBtnClickListemer.OnCancleBtnClick();
            }
        });
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

    }
}

File: SmartPlay/src/main/java/com/luffy/smartplay/player/MyDialog.kt
package com.luffy.smartplay.player

import com.luffy.smartplay.R
import android.app.Dialog
import android.content.*
import android.view.*
import android.widget.*
import java.util.ArrayList

class MyDialog : Dialog, AdapterView.OnItemClickListener {
    private var mListView: ListView? = null
    private var mAdapter: ArrayAdapter<*>? = null
    private var mList: ArrayList<String?>? = null
    private val okBtn: Button? = null
    private val cancleBtn: Button? = null
    private var buttonNum = 2

    //接口对象
    private val mOnOKBtnClickListemer: OnBtnClickListemer? = null

    constructor(context: Context?, buttonNum: Int) : super(context!!, R.style.dialog) {
        this.buttonNum = buttonNum
        val window = this.window
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window!!.setContentView(R.layout.my_dailog_layout)
        initView(window)
    }

    //构造函数
    constructor(context: Context?) : super(context!!, R.style.dialog) {
        val window = this.window
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window!!.setContentView(R.layout.my_dailog_layout)
        initView(window)
    }

    private fun initView(window: Window?) {
        //  okBtn = (Button) window.findViewById(R.id.ok_btn);
        // cancleBtn = (Button) window.findViewById(R.id.cancle_btn);
        if (buttonNum == 1) {
            cancleBtn!!.visibility = View.GONE
        }
        mListView = window!!.findViewById<View>(R.id.id_dialog_list) as ListView
        mListView!!.onItemClickListener = this
        mList = ArrayList()
        mList!!.add(0, "我的收藏")
        mList!!.add(1, "东方幻想")
        mList!!.add(2, "动漫收藏")
//        mAdapter = ArrayAdapter<Any?>(window.context, android.R.layout.simple_list_item_1, mList!!)
        mListView!!.adapter = mAdapter
    }

    fun showDiglog() {
        show()
    }

    fun setBtnText(leftText: String?, rightText: String?) {
        okBtn!!.text = leftText
        cancleBtn!!.text = rightText
    }

    fun setBtnText(leftText: String?) {
        okBtn!!.text = leftText
    }

    fun setBtnClick(onBtnClickListemer: OnBtnClickListemer) {
        okBtn!!.setOnClickListener {
            onBtnClickListemer.OnOKBtnClick() //接口对象的方法，需要在实例化MyDialog是实现该回调方法
        }
        cancleBtn!!.setOnClickListener {
            dismiss()
            onBtnClickListemer.OnCancleBtnClick()
        }
    }

    override fun onItemClick(parent: AdapterView<*>?, view: View, position: Int, id: Long) {}
}